ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/product/Product.java

```diff
@@ -66,7 +69,7 @@ public class Product extends ProductObject implements IProduct {
 
 	private TreeMap<String, IProductObject> fPlugins = new TreeMap<>();
```

Comment:
Can we use a LinkedMap her too?

Reply:
I think the motivation to use a TreeMap is to sort all plug-ins alphabetically according to their symbolic bundle name.
In contrast to Products based on features one cannot move plug-ins of plug-in based products up and down in the editor. So if that is changed there should be some buttons to move plug-ins too. Otherwise one can only add plug-ins at the end of the list (or modify the file directly in the Source tab). But if the order cannot be controlled by the user  having it sorted alphabetically is the best way to make the order arbitrary.

But if you want to be able specify a order of plug-ins I suggest to have that in a dedicated PR (since this is more than just changing the implementation) or discuss it in a separate issue. But I have to say that personally I don't have a great need for that.